pandas/core/dtypes/missing.py

```diff
@@ -754,10 +754,11 @@ def isna_all(arr: ArrayLike) -> bool:
     chunk_len = max(total_len // 40, 1000)
 
     dtype = arr.dtype
-    if dtype.kind == "f":
+    is_np_missing_value = na_value_for_dtype(dtype) is not libmissing.NA
```

Comment:
More specifically, the two `if` blocks below should check "is the dtype's missing value a numpy missing value" since an EA author could define `ExtensionDtype.kind == "f"` and `ExtensionDtype.na_value` as a non-numpy missing value for example.

Should `ArrowDtype.kind == "O"` always be true or is there a better way to specify  "is the dtype's missing value a numpy missing value" cc @jbrockmendel

Reply:
Cool makes sense. Changed it to just check the dtype instead